src/Components/Filters.js

```diff
@@ -1,54 +1,154 @@
-        <Button title="Intermediate" onPress={() => onLevelChange('intermediate')} active={level === 'intermediate'} />
-        <Button title="Advanced" onPress={() => onLevelChange('advanced')} active={level === 'advanced'} />
+    this.onNumberOfPlayersChange = this.onInputChange.bind(this, 'numberOfPlayers');
+  }
+
+  onPressedChange(target, value) {
+    this.setState(prevState => {
+      const newValue = value === prevState[target] ? undefined : value;
+      return {
+        [target]: newValue,
+      };
+    }, this.applyFilters);
+  }
+
+  onInputChange(target, value) {
+    this.setState({ [target]: value }, this.applyFilters);
+  }
+
+  applyFilters() {
+    const { selectedLevel, selectedGoal, numberOfPlayers } = this.state;
+    let newData = this.props.route.params.initialData;
+
+    if (selectedLevel) newData = newData.filter(drill => drill.level === selectedLevel);
+    if (selectedGoal) newData = newData.filter(drill => drill.goals.includes(selectedGoal));
+    if (numberOfPlayers) newData = newData.filter(drill => drill.minimalPlayersNumber <= numberOfPlayers);
+
+    this.setState({ displayedDrills: newData });
+  }
+
+  componentDidMount() {
+    this.props.navigation.setOptions({
+      headerRight: () => (
+        <HeaderButton
+          onPress={() => {
+            this.props.navigation.navigate(this.props.route.params.previousScreen, {
+              filteredDrills: this.state.displayedDrills,
+              type: this.props.route.params.previousType,
+            });
+          }}
+        />
+      ),
+    });
+  }
+
+  render() {
+    const { selectedLevel, selectedGoal, numberOfPlayers } = this.state;
+    return (
+      <View style={styles.wrapper}>
+        <Text style={styles.counter}>{this.state.displayedDrills.length} drills available</Text>
+        <View style={styles.filters}>
+          <Text style={styles.filterTitle}>Level</Text>
+          <View style={styles.filter}>
+            {Object.values(Levels).map(level => (
+              <Button
+                title={level}
+                onPress={() => this.onPressedChange('selectedLevel', level)}
+                key={level}
+                active={selectedLevel === level}
+              />
+            ))}
+          </View>
+          <Text style={styles.filterTitle}>Goals</Text>
+          <View style={styles.filter}>
+            {this.props.route.params.initialData
+              .map(drill => drill.goals)
+              .reduce((x, y) => x.concat(y), [])
+              .filter((goal, index, array) => array.indexOf(goal) === index)
+              .map(goal => (
+                <Button
+                  title={goal}
+                  onPress={() => this.onPressedChange('selectedGoal', goal)}
+                  key={goal}
+                  active={selectedGoal === goal}
+                />
+              ))}
+          </View>
+          <Text style={styles.filterTitle}>Number of players: {numberOfPlayers}</Text>
+          <Slider
```

Comment:
I do not why but on my phone, when I use the slider. The animation does not work well when I keep my finger on it

Reply:
Weird...

I was planning to move to another library with the fitness filters to be able to choose a range for the duration of drills. (maybe https://github.com/ptomasroos/react-native-multi-slider)

Moreover Slider is a deprecated react-native feature, but the up-to-date library is not yet available in expo apps. I think we can keep a "not perfect" slider for now and move quickly hen a better on is available. More info there : https://github.com/react-native-community/react-native-slider/issues/152#issuecomment-593786774